x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/esql/action/EsqlQueryResponse.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.esql.action;
+
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.core.Releasable;
+
+import java.util.Iterator;
+import java.util.List;
+
+public abstract class EsqlQueryResponse extends ActionResponse implements Releasable {
+
+    public abstract List<? extends ColumnInfo> columns();
```

Comment:
Since we'll have pagination at some point (whether exposed or not), how about extracting these methods separate from the EsqlQueryResponse:
```
public abstract class EsqlQueryResponse extends ActionResponse implements Releasable {
   public EsqlResponse response();
}

public abstract class EsqlResponse {
   public abstract List<ColumnInfo> columns();
   public abstract Iterable<Iterable<Object>> rows();
   public  abstract Iterable<Object> column(int index);
}
```

Reply:
yes. I like it. I also like the _rows_ and _column_ abstraction - lemme see how it'll fit.

Reply:
Done.